ucdlib-directory/includes/people.php

```diff
@@ -283,7 +300,180 @@ public function register_post_meta(){
       'default' => false,
       'type' => 'boolean',
     ) );
+  }
+  
+  // ensure block exists
+  public function ensureBlockExists($block_name, $dry_run=false, $inCLI=false, $insert_after_block='') {
+    $post_type = $this->slug; 
+    if($inCLI) echo "\n== Directory People: Ensure {$block_name} block ==";
+
+    if (empty($block_name)) {
+      if($inCLI) echo "\n Block Name is empty, please give a valid blockname'.";
+      else {
+          $blockNameMessage = wp_json_encode("Block Name is empty, please give a valid blockname");   
+          echo "<script>console.warn({$blockNameMessage});</script>";
+      }
+      return;
+    }
+
+    if (empty($post_type)) {
+      if($inCLI) {
+        echo "\n ERROR: people post type slug is empty.\n";
+      }
+      else {
+          $postWarnMessage = wp_json_encode("{$post_type} slug is empty.");   
+          echo "<script>console.warn({$postWarnMessage});</script>";
+      }
+      return;
+    }
 
+    $q = new WP_Query([
+      'post_type'      => $post_type,
+      'post_status'    => 'any',
+      'posts_per_page' => -1,
+      'fields'         => 'ids',
+      'no_found_rows'  => true,
+    ]);
+
+    if($inCLI) echo "\n Found {$q->post_count} {$post_type} posts.";
+
+    // Filter out any invalid/empty IDs so we only process real posts
+    $valid_ids = array_filter($q->posts, function($id) {
```

Comment:
by setting `'post_status' => 'publish'` in your query, you can completely eliminate this line. - and then you don't have to do an extra db query (`get_post_status`) for each post.